src/PartitionLoops.cpp

```diff
@@ -519,8 +519,30 @@ class PartitionLoops : public IRMutator {
     bool in_gpu_loop = false;
 
     Stmt visit(const For *op) override {
+        // Do not partition if the schedule explicitly forbids.
+        if (op->partition_policy == LoopPartitionPolicy::Never) {
+            return op;
```

Comment:
Doesn't this need to recursively visit the child stmt to partition inner loops? I think this should be `return IRMutator::visit(op);`

Reply:
Wauw, good catch! Tricky and subtle one. Might have caused some headaches :smiling_face_with_tear: